components/Menu/Menu.js

```diff
@@ -2,19 +2,29 @@ import { useEffect, useState } from "react";
 import MenuIcon from "@/public/icons/Menu.svg";
 import MenuItem from "../MenuItem/MenuItem";
 import styled, { keyframes, css } from "styled-components";
+import Login from "../Login/Login";
+import { useSession } from "next-auth/react";
 
 export default function Menu({
   collections,
   startClosingForm,
   changeFlashcardSelection,
   getAllFlashcardsFromCollection,
 }) {
+  const { data: session } = useSession();
+
   const [isMenu, setIsMenu] = useState(false);
   const [isMenuClosing, setIsMenuClosing] = useState(false);
 
   const [submenuMode, setSubmenuMode] = useState("default");
   const [closingTrigger, setClosingTrigger] = useState(false);
 
+  const [showLogOutDialog, setShowLogOutDialog] = useState(false);
+
+  function changeShowLogOutDialog(value) {
+    setShowLogOutDialog(value);
+  }
```

Comment:
If I am not mistaken, this state is only used in both Login Components. You can move this state inside that component and reduce the complexity.